src/java.base/share/classes/java/lang/StringConcatHelper.java

```diff
@@ -319,6 +345,47 @@ static long prepend(long indexCoder, byte[] buf, String value, String prefix) {
         return indexCoder;
     }
 
+    /**
+     * Prepends the stringly representation of FormatConcatItem value into buffer,
+     * given the coder and final index. Index is measured in chars, not in bytes!
+     *
+     * @param indexCoder final char index in the buffer, along with coder packed
+     *                   into higher bits.
+     * @param buf        buffer to append to
+     * @param value      String value to encode
+     * @return           updated index (coder value retained)
+     * @since 19
+     */
+    @PreviewFeature(feature=PreviewFeature.Feature.STRING_TEMPLATES)
```

Comment:
Just a passerby question: what's the point in marking non-public methods with `@PreviewFeature`?

Reply:
It also lets JDK developers know that the methods are transitional.